subxt/src/backend/mod.rs

```diff
@@ -325,9 +325,417 @@ pub enum TransactionStatus<Hash> {
+            ];
+            let rpc_client = build_mock_client(mock_data, vec![]).await;
+
+            // Test
+            let backend: LegacyBackend<Conf> = LegacyBackend::builder().build(rpc_client);
+            let response = backend.genesis_hash().await.unwrap();
+
+            assert_eq!(hash, response)
+        }
+
+        #[tokio::test]
+        /// This test should cover following methods:
+        /// - `stream_runtime_version`
+        /// - `stream_all_block_headers`
+        /// - `stream_best_block_headers`
+        /// The test covers them because they follow the simple pattern of:
+        /// ```no_run
+        /// async fn stream_the_thing(
+        ///     &self,
+        /// ) -> Result<StreamOfResults<(T::Header, BlockRef<T::Hash>)>, Error> {
+        ///     let methods = self.methods.clone();
+        ///    let retry_sub = retry_stream(move || {
+        ///         let methods = methods.clone();
+        ///         Box::pin(async move {
+        ///               methods.do_the_thing().await?
+        ///             });
+        ///             Ok(StreamOf(Box::pin(sub)))
+        ///         })
+        ///     })
+        ///     .await?;
+        ///     Ok(retry_sub)
+        /// }
+        /// ```
+        async fn stream_simple() {
+            // Setup
+
+            let mock_subscription_data = vec![
+                Ok(vec![
+                    Ok(runtime_version(0)),
+                    Err(RpcError::DisconnectedWillReconnect(
+                        "Reconnecting".to_string(),
+                    )),
+                    Ok(runtime_version(1)),
+                ]),
+                Ok(vec![
+                    Err(RpcError::DisconnectedWillReconnect(
+                        "Reconnecting".to_string(),
+                    )),
+                    Ok(runtime_version(2)),
+                    Ok(runtime_version(3)),
+                ]),
+                Ok(vec![
+                    Ok(runtime_version(4)),
+                    Ok(runtime_version(5)),
+                    Err(RpcError::RequestRejected("Reconnecting".to_string())),
+                ]),
+            ];
+            let rpc_client = build_mock_client(vec![], mock_subscription_data).await;
+
+            // Test
+            let backend: LegacyBackend<Conf> = LegacyBackend::builder().build(rpc_client);
+
+            let mut results = backend.stream_runtime_version().await.unwrap();
+            let mut expected = VecDeque::from(vec![
+                Ok::<crate::client::RuntimeVersion, crate::Error>(client_runtime_version(0)),
+                Ok(client_runtime_version(4)),
+                Ok(client_runtime_version(5)),
+            ]);
+
+            while let Some(res) = results.next().await {
+                if res.is_ok() {
+                    assert_eq!(expected.pop_front().unwrap().unwrap(), res.unwrap())
+                } else {
+                    assert!(matches!(
```

Comment:
Ok, this special assert is because the Error doesn't impl PartialEq?

Reply:
Yup, deriving it all the way through is a bit painful